src/format_bench.rs

```diff
@@ -0,0 +1,76 @@
+
+use criterion;
+
+pub fn big_table(b: &mut criterion::Bencher, size: &usize) {
+    let mut table = Vec::with_capacity(*size);
+    for _ in 0..*size {
+        let mut inner = Vec::with_capacity(*size);
+        for i in 0..*size {
+            inner.push(i);
+        }
+        table.push(inner);
+    }
+    let mut table_str = String::new();
+    for r1 in table {
+        table_str.push_str("<tr>\n");
+        for r2 in r1 {
+            table_str.push_str(format!("<td>{col}</td>",col = r2).as_str());
+        }
+        table_str.push_str("</tr>\n");
+    }
+    let output = format!(
+"<table>
+{table_str}
+</table>", table_str = table_str);
+
+    b.iter(|| output.to_owned());
```

Comment:
This does not actually benchmark the `format!()` macro. For that to happen, I think you have to make the construction of `table_str` (so line 13-24) part of the closure passed to `b.iter()`.